h2o-core/src/main/java/water/rapids/ast/prims/assign/AstRectangleAssign.java

```diff
@@ -265,22 +235,37 @@ private void assign_frame_scalar(Frame dst, int[] cols, AstNumList rows, String
     for (int i = 0; i < cols.length; i++)
       vecs2[i] = vecs[cols[i]];
     rows.sort();                // Side-effect internal sort; needed for fast row lookup
-    new AssignFrameStringScalarTask(rows, src).doAll(vecs2);
+    AssignFrameScalarTask.doAssign(rows, vecs2, src);
   }
 
-  private static class AssignFrameStringScalarTask extends RowSliceTask {
-    private String _src;
-    private AssignFrameStringScalarTask(AstNumList rows, String src) {
+  private static class AssignFrameScalarTask extends RowSliceTask {
```

Comment:
Yes, but code in rapids should not be making decisions about what should be the representation of the chunk. This should be responsibility of the chunk layer. We could introduce a new chunk method set(fromIndex, toindex, value) - this method could make use of the information that an interval is constant and figure out the best chunk representation.

I don't think we currently have implementation of a constant string chunk.